### ISSUE TITLE
Circular Reference with Generic Parameterization -  error TS2456: Type alias 'Result' circularly references itself.

### ISSUE THREAD

**Issue by AaronNGray** [microsoft/TypeScript#61062]
### 🔎 Search Terms

Circular Reference, Generic Parameterization, error TS2456: Type alias 'Result' circularly references itself.

### 🕗 Version & Regression Information

Failing on Main 32ebd28578902d5437e541cf185ea2a32742686d

### ⏯ Playground Link

https://www.typescriptlang.org/play/?#code/C4TwDgpgBAggdiAPAFShAHsCcAmBnKAeQCMArCAY2AD4oBeKAbwCgo2oBtAawhAC48wAE4BLOAHMAunyjIA3MwC+cqM2ahIUAEoQ8AVwA2welEGiJUAD7bdh4B0lXYCRDv1HqKgPRe0QoQD2QrIAygBMACwArABsMsjg0ACGBiJJBADkbnYZUBQiQhSGSUIGIFBCEABmEJVwFLpQIsB4EAZVAHTMQA

### 💻 Code

Fails :-
```ts
type Any<T extends Object> = {
    [key:string]: T;
}; 

type Result = string | Result[] | Any<Result>; // error TS2456: Type alias 'Result' circularly references itself.
```
Works :-
```
type ResultObject = {
    [key:string]: Result;
}; 

type Result = string | Result[] | ResultObject;
```
Fails :-
```
type ResultObject = {
    [key:string]: Result;
}; 

type ResultArray<T> = T[];
 
type Result = string | ResultArray<Result> | ResultObject;  // error TS2456: Type alias 'Result' circularly references itself.
```
It looks like it is a Generic Parameterization issue.

### 🙁 Actual behavior

Throwing Error on compilation error TS2456: Type alias 'Result' circularly references itself.

### 🙂 Expected behavior

Should not throw an error and demonstrates the type system is incomplete.

**Comment by jcalz** [comment]
duplicate of #41164

**Comment by AaronNGray** [comment]
Looks semantically totally different to me.

**Comment by jcalz** [comment]
`Any<T>` is the same as `Record<string, T>` and you have the same problem with it.

**Comment by AaronNGray** [comment]
So basically there is no support for recursive parameterized types and mutually recursive parameterized types in TypeScript.

This is a crazy limitation as so many programming patterns I use use this.

I need to find my feet with TypeScripts code base and see if I can implement this.

**Comment by jcalz** [comment]
You might want to read through the other issue thoroughly, especially the linked PR at #57293.
